### PR TITLE
feat(v2): add option validation for remaining official plugins

### PR DESCRIPTION
**Pull request by teikjun**
## Motivation

This PR provides option validation for the remaining official plugins. 

This is a follow-up PR to #2943. #2943 generalizes option validation for plugins and themes. This PR uses the same pattern to validate and normalize options for the remaining official plugins.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Tests have been added for the respective plugins.

## Related PRs

This PR should be merged after #2943. I will pull the changes after #2943 is merged to resolve any merge conflicts first. (Edit: done)

### CODE REVIEW

File: packages/docusaurus-plugin-content-pages/src/pluginOptionSchema.ts [hunk 1]
```diff
@@ -5,9 +5,16 @@
  * LICENSE file in the root directory of this source tree.
  */
 import * as Joi from '@hapi/joi';
+import {PluginOptions} from './types';
+
+export const DEFAULT_OPTIONS: PluginOptions = {
+  path: 'src/pages', // Path to data on filesystem, relative to site dir.
+  routeBasePath: '', // URL Route.
+  include: ['**/*.{js,jsx,ts,tsx}'], // Extensions to include.
+};
```
[the remark this answers is not in the document]
Comment (anshulrgoyal) [reply packages/docusaurus-plugin-content-pages/src/pluginOptionSchema.ts:14]: yes look more verbose. I just changed my pr to follow this

**Comment by slorber** [comment]
Hey, is this ready to review/merge?

The other pr has been merged: https://github.com/facebook/docusaurus/pull/2943

**Comment by anshulrgoyal** [comment]
It is almost ready, we need to rebase it and make some minor changes

**Comment by teikjun** [comment]
Ready for review now :D

**Comment by slorber** [comment]
there's a conflict :p

**Comment by teikjun** [comment]
conflict resolved! :)

**Comment by slorber** [comment]
great job, let's merge this 👍

**Pull request merged**